src/main/java/net/snowflake/client/jdbc/SnowflakeFileTransferAgent.java

```diff
@@ -3101,28 +3287,56 @@ public static void throwJCEMissingError(String operation, Exception ex)
       logger.error(msg);
     }
     throw new SnowflakeSQLException(
-        ex, SqlState.SYSTEM_ERROR, ErrorCode.AWS_CLIENT_ERROR.getMessageCode(), operation, msg);
+        queryId,
+        ex,
+        SqlState.SYSTEM_ERROR,
+        ErrorCode.AWS_CLIENT_ERROR.getMessageCode(),
+        operation,
+        msg);
   }
 
   /**
    * For handling IOException: No space left on device when attempting to download a file to a
    * location where there is not enough space. We don't want to retry on this exception.
    *
+   * @deprecated use {@link #throwNoSpaceLeftError(SFSession, String, Exception, String)}
    * @param session the current session
    * @param operation the operation i.e. GET
    * @param ex the exception caught
    * @throws SnowflakeSQLLoggedException
    */
+  @Deprecated
   public static void throwNoSpaceLeftError(SFSession session, String operation, Exception ex)
       throws SnowflakeSQLLoggedException {
+    throwNoSpaceLeftError(session, operation, ex, null);
+  }
+
+  /**
+   * For handling IOException: No space left on device when attempting to download a file to a
+   * location where there is not enough space. We don't want to retry on this exception.
+   *
+   * @param session the current session
+   * @param operation the operation i.e. GET
+   * @param ex the exception caught
+   * @throws SnowflakeSQLLoggedException
+   */
+  public static void throwNoSpaceLeftError(
+      SFSession session, String operation, Exception ex, String queryId)
+      throws SnowflakeSQLLoggedException {
     String exMessage = SnowflakeUtil.getRootCause(ex).getMessage();
     if (exMessage != null && exMessage.equals(NO_SPACE_LEFT_ON_DEVICE_ERR)) {
       throw new SnowflakeSQLLoggedException(
+          queryId,
           session,
           SqlState.SYSTEM_ERROR,
           ErrorCode.IO_ERROR.getMessageCode(),
           ex,
           "Encountered exception during " + operation + ": " + ex.getMessage());
     }
   }
+
+  private static String queryIdFromEncMatOr(
```

Comment:
Is it the same as in QueryIdHelper?

Reply:
it's the same but QueryIdHelper is in a different package and I don't want to make it public

additionally in this class it was the only one usage for this method so I inlined it